### PR TITLE
Cascade ts defs

### PR DESCRIPTION
**Pull request by aParadowski**
- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/master/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/material-ui/src/MenuItem/MenuItem.d.ts [hunk 1]
```diff
@@ -10,6 +11,43 @@ export type MenuItemTypeMap<P = {}, D extends React.ElementType = 'li'> = Omit<
...
+     * Fires when the right arrow key is pressed
+     * on a MenuItem that contains nested items
+     * and passes focus to the first child of the
+     * nested items array
+     */
+    handleArrowRightKeydown?: React.ReactEventHandler<{}>;
+    /**
+     * An array of MenuItems to render in a nested Menu
+     */
+    nestedItems?: React.ReactNode;
```
Comment (EsoterikStare) [inline packages/material-ui/src/MenuItem/MenuItem.d.ts:25]: This should be an array of components. I think the syntax looks like this (but not 100% sure):

```suggestion
    nestedItems?: React.ReactNode[];
```

Comment (aParadowski) [reply packages/material-ui/src/MenuItem/MenuItem.d.ts:25]: Not sure about that, Menu's definition for children is `children?: React.ReactNode;`

Comment (aParadowski) [reply packages/material-ui/src/MenuItem/MenuItem.d.ts:25]: https://flow.org/en/docs/react/types/#toc-react-node

Comment (EsoterikStare) [reply packages/material-ui/src/MenuItem/MenuItem.d.ts:25]: Huh. Interesting.

**Pull request merged**